Build fails when using source download

<!-- Have you done the following? -->
<!--   * read the Code of Conduct? By filing an Issue, you are expected to -->  
<!--     comply with it, including treating everyone with respect: -->
<!--     https://github.com/ConsenSys/teku/blob/master/CODE-OF-CONDUCT.md -->
<!--   * Reproduced the issue in the latest version of the software -->
<!--   * Read the debugging wiki: https://github.com/ConsenSys/teku/wiki/debugging -->
<!--   * Duplicate Issue check:  https://github.com/search?q=+is%3Aissue+repo%3AConsenSys/Teku -->
<!-- Note:  Not all sections will apply to all issue types. -->

### Description
Teku 21.9.1 release build failure

### Acceptance Criteria
* [Criteria 1]

### Steps to Reproduce (Bug)
1. freshly installed Ubuntu 20.04
2. sudo apt install openjdk-11-jdk-headless
3. download and extract teku 21.9.1 release
4. ./gradlew installDist

**Expected behavior:** 
build succeeds

**Actual behavior:**
```
teku-21.9.1$ ./gradlew --version
------------------------------------------------------------
Gradle 7.0.2
------------------------------------------------------------

Build time:   2021-05-14 12:02:31 UTC
Revision:     1ef1b260d39daacbf9357f9d8594a8a743e2152e

Kotlin:       1.4.31
Groovy:       3.0.7
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          11.0.11 (Ubuntu 11.0.11+9-Ubuntu-0ubuntu2.20.04)
OS:           Linux 5.4.0-86-generic amd64

teku-21.9.1$ ./gradlew installDist --warning-mode=all --info
...
FAILURE: Build failed with an exception.

* Where:
Build file '/home/jerryj/download/teku-21.9.1/build.gradle' line: 757

* What went wrong:
A problem occurred evaluating root project 'teku'.
> toIndex = 2
```

**Frequency:**
100%

### Versions (Add all that apply)
* Software version: 21.9.1 release
* Java version:
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment (build 11.0.11+9-Ubuntu-0ubuntu2.20.04)
OpenJDK 64-Bit Server VM (build 11.0.11+9-Ubuntu-0ubuntu2.20.04, mixed mode, sharing)

* OS Name & Version:
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.3 LTS"

* Kernel Version: Linux 5.4.0-86-generic #97-Ubuntu SMP Fri Sep 17 19:19:40 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
* Virtual Machine software & version: NA
* Docker Version: NA
* Cloud VM, type, size: 4 CPU, 8G RAM

### Additional Information

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -29,3 +29,5 @@ For information on changes in released versions of Teku, see the [releases page]
 ### Bug Fixes
  - Fixed a possible crash on shutdown when using levelDb.
  - Set an idle timeout for metrics connections, to clean up ports when no longer used
+ - Fixed error when building from a source download rather than a git checkout.
+   Now logs a warning when building without git information to warn users that version information will not be available.

--- a/build.gradle
+++ b/build.gradle
@@ -702,7 +702,7 @@ def calculatePublishVersion() {
 // Otherwise use git describe --tags and replace the - after the tag with a +
 def calculateVersion() {
   if (!grgit) {
-    logger.warn("Not building from a git checkout. Version information will not be available. Checkout out source from git is strongly recommended.")
+    logger.warn("Not building from a git checkout. Version information will not be available. Building from a git checkout is strongly recommended.")
     return 'UNKNOWN+develop'
   }
   String version = grgit.describe(tags: true)

--- a/build.gradle
+++ b/build.gradle
@@ -687,7 +687,7 @@ def buildTime() {
 // Otherwise, use develop
 def calculatePublishVersion() {
   if (!grgit) {
-    return 'UNKNOWN'
+    return 'develop'
   }
   def specificVersion = calculateVersion()
   def isReleaseBranch = grgit.branch.current().name.startsWith('release-')
@@ -702,7 +702,8 @@ def calculatePublishVersion() {
 // Otherwise use git describe --tags and replace the - after the tag with a +
 def calculateVersion() {
   if (!grgit) {
-    return 'UNKNOWN'
+    logger.warn("Not building from a git checkout. Version information will not be available. Checkout out source from git is strongly recommended.")
+    return 'UNKNOWN+develop'
   }
   String version = grgit.describe(tags: true)
   if (version == null) {

